### PR TITLE
Fix unstable embed template literals

### LINKED ISSUE #6150: Unstable template literals marked with /* HTML */ and /* GraphQL */
Happens if the template literal initially doesn't contain line breaks.

Another issue about unstable template literals: #5885

### HTML in JS ###

**Prettier 1.17.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzCEAUB6AVAAgAkAVAWQBl9dt8ADAHgEIBaZ-AIwEMAnARn1YA+Dj3xNWI7gCYBzQbQCUAbhAAaEBAAOMAJbQAzslA9uEAO4AFHgkMpOANwg6AJmpDtunMAGs4MAMqaXjpQAObIMNwArnDqABYwALYANgDqcTrw+kFgcP42mTr2mQCeyOD6huoh+nDcMBaeoYmcyKicybXqAFb6AB4AQp4+fv6ciXDkIXBtHV0gvX3+IaHJcACKURDws52xIEHctdzlXOwlydBumtwhMKkuMHHIABwADOo3ELWpnprlNzgx3sM3UAEctvBGlpbCBOPpmFA4HBnCi3Nw4BCdBjGpxmq0kO09upaokdBFovt9Cs1pttjNCXN9jBOOwHs4nsgpOpIpwdMkVgBhCCJFrlKDQUEgKK1Yis2xE2oAXyVQA)
```sh
--parser babylon
```

**Input:**
```jsx
foo(/* HTML */ `<!-- bar1 --> bar <!-- bar2 -->`);
```

**Output:**
```jsx
foo(
  /* HTML */ `
    <!-- bar1 -->
    bar
    <!-- bar2 -->
  `
);

```

**Second Output:**
```jsx
foo(/* HTML */ `
  <!-- bar1 -->
  bar
  <!-- bar2 -->
`);

```

### GraphQL in JS ###

**Prettier 1.17.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzCEAUB6AVAAgHEAnAQwAcALARQBl9dt8ADARwFc5iBPfYfdBD74ARqWL4AvlOYBKANwgANCAjkYAS2gBnZKHHEIAdwAK4hLpSkAbhA0ATZSBFkwAazgwAyuVJgNUADmyDDEnCqUMAC2ADYA6pQa8Nq+YHBeFkka1kncyODauioB2lwwJmSBUaTIqKQxpSoAVtoAHgBCrh7epFFwtAFwtfWNIC2tXgGBMXDU7BDwww1wKr7EpcT5YiLcMdBO5MQBMHEOMJTIABwADKuGpXFk5PmHcBvWQyocC3AVapYgUjaAC0UDgcHsEKcxDgHA0MIqpCqNSQdWWKlKUQ0ITCKxA2imMzmPyWoxgpBEp3s52QACYVKFSBoYlMAMIQKLVfJQaCfEDsUoAFQpljRpUkkiAA)
```sh
--parser babylon
```

**Input:**
```jsx
foo(/* GraphQL */ `query { foo { bar } }`);
```

**Output:**
```jsx
foo(
  /* GraphQL */ `
    query {
      foo {
        bar
      }
    }
  `
);

```

**Second Output:**
```jsx
foo(/* GraphQL */ `
  query {
    foo {
      bar
    }
  }
`);

```

### PR DESCRIPTION
**Pull request by fisker**
As posted in https://github.com/prettier/prettier/issues/6150#issuecomment-607172797, we can't know tagged template will break or not, let's treat it like a normal argument, if we found some case ugly, we can improve in different way.

Fixes #6150

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/js/multiparser-css/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -285,13 +285,12 @@ export const Group = styled.div\`
...
-  \${ChatRoot}[data-frame="yes"] & {
-    \${css({}, ...args)}
-  }
-\`;
+const yesFrame = (...args: Interpolation<ThemedStyledProps<{}, Theme>>[]) =>
+  css\`
+    \${ChatRoot}[data-frame="yes"] & {
+      \${css({}, ...args)}
+    }
+  \`;
```
Comment (evilebottnawi) [inline tests/js/multiparser-css/__snapshots__/jsfmt.spec.js.snap:293]: Why it was changed?

Comment (fisker) [reply tests/js/multiparser-css/__snapshots__/jsfmt.spec.js.snap:293]: Before, if it's ``` css`` ``` and contains new lines, it print `css` on same line of arrow function. But problem is if ``` css`` ``` formatted as single line, the second format will change to print ``` css`` ``` on new line, so this became unstable.

File: tests/js/template/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -130,7 +136,8 @@ var o = {
 };
 
 var x = {
-  y: () => Relay.QL\`
+  y: () =>
+    Relay.QL\`
```
Comment (thorn0) [inline tests/js/template/__snapshots__/jsfmt.spec.js.snap:140]: I don't think this is desired. These tests are completely unrelated to the issue being fixed, so they shouldn't change.

Comment (fisker) [reply tests/js/template/__snapshots__/jsfmt.spec.js.snap:140]: The old one, if you insert a empty line before the `Relay.QL`, it format like this too, do we really want distinguish?

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEA3AhgJwAQA9sC82wAOlNtgJ5LYAUAlIQHzZlkXutQBKcANukoA6AIoAZAAacKARwCucTJWLSK2ACTAAZhAgBfADSqKWgJb8AJkfJq9nKVD0BuEAZAQADjFPQAzslAsTAgAdwAFLAR-FHQ+EMF-NwAjTHQwAGs4GABlDzTTKABzZBhMBTcACxgAWz4AdQrTeF88sDhsqKbTVCbKZHBfRJAC30UYMNTC6vRkLVjRtwArX1wAIVSMrOz0argxArhZ+bglleyCwr44ETkIeCO+BZA8zFHMfqT0JP5XZ8wCmB1UwWGAVZAADgADG4PMFRnVUh5+rC4G9UIc3PI7nAJp5oiB0L4ALRQOBwCzk36YODyUzUiboKYzJBzR4nECjaqmEpldm+C5XG7Yh5PGBfIEgsFIABMblK6FMfAuAGEINVpv1UQBWX5yUYAFS+0VZT1QCgAklAKbBsmB-l4AIJW7IwShXEVwPR6IA)
```sh
--parser babel
```

**Input:**
```jsx
var x = {
  y: () => 

  
  
Relay.QL`
    query {
      ${foo},
      field,
    }
  `
};
```

**Output:**
```jsx
var x = {
  y: () =>
    Relay.QL`
    query {
      ${foo},
      field,
    }
  `,
};

```

Comment (thorn0) [reply tests/js/template/__snapshots__/jsfmt.spec.js.snap:140]: Unfortunately, yes. I don't like it, but this is intentional. You can use git blame to find the respective PRs and discussion.

**Comment by thorn0** [comment]
superseded by https://github.com/prettier/prettier/pull/13532

File: tests/js/template/__snapshots__/jsfmt.spec.js.snap (hunk 2 above)
Comment (thorn0) [reply tests/js/template/__snapshots__/jsfmt.spec.js.snap]: For reference, that PR was https://github.com/prettier/prettier/pull/1497
I'd still like to find a way to eliminate this dependency on the input formatting.
